packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -326,6 +334,34 @@ export const useGridSelection = (
     [apiRef, props.checkboxSelectionVisibleOnly, props.pagination],
   );
 
+  const handleCellKeyDown = React.useCallback<GridEventListener<GridEvents.cellKeyDown>>(
+    (params, event) => {
+      // The target is not an element when triggered by a Select inside the cell
+      // See https://github.com/mui-org/material-ui/issues/10534
+      if ((event.target as any).nodeType === 1 && !isGridCellRoot(event.target as Element)) {
+        return;
+      }
+
+      // Get the most recent params because the cell mode may have changed by another listener
+      const cellParams = apiRef.current.getCellParams(params.id, params.field);
+      const isEditMode = cellParams.cellMode === GridCellModes.Edit;
+      if (isEditMode) {
+        return;
+      }
+
+      if (isSpaceKey(event.key) && event.shiftKey) {
```

Comment:
Since this is new code, let's avoid these one line helpers.

```suggestion
      if (event.key === ' ' && event.shiftKey) {
```